File: python/questions/sortedSquares.py
from typing import List

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        # r = len(nums) - 1
        # l = 0
        # ans = [0] * len(nums)
        # for p in range(len(nums)-1, -1, -1):
        #     if nums[r] > -nums[l]:
        #         ans[p] = nums[r] * nums[r]
        #         r -= 1
        #     else:
        #         ans[p] = nums[l] * nums[l]
        #         l += 1
        # return ans
        return sorted([num**2 for num in nums])

File: cpp/questions/sortedSquares.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l = 0;
        int r = nums.size() - 1;
        if (r == 0) {
            nums[0] *= nums[0];
            return nums;
        }
        std::vector<int> ans(nums.size(), 0);
        nums[l] *= nums[l];
        nums[r] *= nums[r];
        for (int p=r; p>1; --p) {
            if (nums[r] > nums[l]) {
                ans[p] = nums[r];
                --r;
                nums[r] *= nums[r];
            } else {
                ans[p] = nums[l];
                ++l;
                nums[l] *= nums[l];
            }
        }
        ans[1] = std::max(nums[l], nums[r]);
        ans[0] = std::min(nums[l], nums[r]);
        return ans;
    }
};
